ui/src/app/components/admin/audit-page-component.tsx

```diff
@@ -50,14 +50,8 @@ export interface AuditPageProps {
 
 const UserInput = ({initialAuditSubject, auditSubjectType, getNextAuditPath, buttonLabel, queryText}) => {
   const [auditSubject, setAuditSubject] = useState(initialAuditSubject);
-  const [loadNextSubject, setLoadNextSubject] = useState(false);
   const [downloadSqlFile, setDownloadSqlFile] = useState(false);
-
-  useEffect(() =>  {
-    if (loadNextSubject) {
-      setLoadNextSubject(false);
-    }
-  }, [loadNextSubject]);
+  const [nextSubjectRequested, setNextSubjectRequested] = useNavigationState();
```

Comment:
Can we just call this `subjectRequested`? The first  one is also a next one, so I don't know that `next`  really signifies anything.

Also,  do we really need this other bit? It seems like we've got a boolean tracking whether another property is non-null and/or changed. Can we not express  what we need with `useEffect` on `auditSubject`?

Reply:
I updated the name

The auditSubject state is needed to update the value in the text box, otherwise it would never change. The boolean state for navigation is more to check whether the user clicked on the "Audit" button. Without that information we have not way of knowing whether the user intended to run a new audit.

The additional state is a different paradigm from our previous navigation - which was basically a straight up callback. This is where react-router's philosophy leaks out a bit. In react-router's view of the world, it wants everything to be a react component, including navigation. In order navigate, we need to render a component.

Reply:
OK, so a purist could have one piece of state to manage/describe the input box's contents, and  update  the next subject when that was committed, but that's no simpler than what you have here.